analyses/snv-callers/util/tmb_functions.R

```diff
@@ -72,92 +74,74 @@ snv_ranges_filter <- function(maf_df, keep_ranges = NULL, bp_window = 0) {
     keep_ranges,
     maxgap = bp_window
   )
-  
+
   # Calculate of ratio of variants in this BED using the @from slot which
-  # indicates the indices of the ranges in `wxs_maf_ranges` that have overlaps
-  # with `wxs_bed_ranges`
+  # indicates the indices of the ranges in `maf_ranges` that have overlaps
+  # with `keep_ranges`
   ratio <- length(overlap@from) / nrow(maf_df)
-  
+
   # What fraction of mutations are in these bed regions?
   cat(
     "Ratio of variants in this BED:", ratio, "\n",
     "Ratio of variants being filtered out:", 1 - ratio, "\n"
   )
-  
+
   # Only keep those in the BED regions that overlap the `wxs_bed_granges`
   filt_maf_df <- maf_df[unique(overlap@from), ]
-  
+
   # Return this matrix with the WXS mutations filtered but WGS the same
   return(filt_maf_df)
 }
 
-calculate_tmb <- function(maf_df, bed_wgs, bed_wxs) {
-  # Calculate Tumor Mutational Burden for each sample given their WGS or WXS
-  # sizes in bp. Based on the BED files provided, tthis function filters out SNV 
-  # outside those ranges and uses the total size in bp of those BED ranges for th
-  # TMB denominator. 
+calculate_tmb <- function(tumor_sample_barcode = NULL,
+                          maf_df,
+                          bed_ranges) {
+  # Calculate Tumor Mutational Burden a given sample in `Tumor_Sample_Barcode`
+  # given a target region BED frame. This function uses `snv_ranges_filter` to
+  # filter out SNVs outside those target BED ranges and uses the total size in
+  # bp of those BED ranges for the TMB denominator.
   #
-  # TMB = # variants / size of the genome or exome surveyed
+  # TMB = # variants / size of the genome or exome surveyed / Mb
   #
   # Args:
+  #   tumor_sample_barcode: A single character string corresponding to a sample
+  #                         id in the `Tumor_Sample_Barcode` MAF file in maf_df.
   #   maf_df: maf data.frame that has been turned into a data.frame, has had
   #           the experimental_stategy column added from the metadata (can be
   #           done with the `set_up_maf` function) and has WXS mutations filtered
   #           using `wxs_bed_filter` function (If the situation calls for it).
-  #   bed_wgs: BED file path with genome ranges to be used for filtering and 
-  #            genome size denominator for WGS samples
-  #   bed_wxs: BED file path genome ranges to be used for filtering and genome 
-  #            size denominator for WXS samples
+  #   bed_df: BED formatted data.frame of the target regions to use for TMB
+  #           calculations.
   #
   # Returns:
-  # A sample-wise data.frame with Tumor Mutation Burden statistics calculated
-  # using the given WGS and WXS sizes.
-  
-  # Read in the BED files we need
-  bed_wgs <- readr::read_tsv(bed_wgs, col_names = FALSE)
-  bed_wxs <- readr::read_tsv(bed_wxs, col_names = FALSE)
-  
+  # A calculated genome size, and TMB for the given Tumor_Sample_Barcode,
```

Comment:
Calling this genome size is potentially confusing.
```suggestion
  # A calculated total region size, and TMB for the given Tumor_Sample_Barcode,
```